assets/src/edit-story/components/library/panes/media/virtualList.js

```diff
@@ -0,0 +1,332 @@
+    onScrollExec(evt);
+  }, 100); //66
+
+  useEffect(() => {
+    const outerNode = outerRef.current;
+    // outerNode.addEventListener('scroll', handleRAFScroll, { passive: true });
+    outerNode.addEventListener('scroll', debouncedHandleScroll, {
+      passive: true,
+    });
+    return () => {
+      // outerNode.removeEventListener('scroll', handleRAFScroll, { passive: true });
+      outerNode.removeEventListener('scroll', debouncedHandleScroll, {
+        passive: true,
+      });
+    };
+  }, [debouncedHandleScroll]);
+
+  useEffect(() => {
+    const outerNode = outerRef.current;
+    const innerNode = innerRef.current;
+    const { scrollTop, clientHeight } = outerNode;
+    setScrollInfo((info) => ({
+      ...info,
+      from: scrollTop,
+      to: scrollTop + clientHeight,
+      height: innerNode.clientHeight,
+    }));
+  }, [resources]);
+
+  const [columns, columnsCumulativeTop] = useMemo(() => {
+    if (resources.length === 0) {
+      return [[], [[], []]];
+    }
+    const columns = [[], []];
+    const columnsCumulativeTop = [[0], [0]];
+
+    const ITEM_MARGIN_BOTTOM = 10;
+
+    for (let i = 0, n = resources.length; i < n; i++) {
+      const resource = resources[i];
+      const { width: originalWidth, height: originalHeight } = resource;
+
+      const oRatio =
+        originalWidth && originalHeight ? originalWidth / originalHeight : 1;
+      const itemHeight = PREVIEW_SIZE / oRatio;
+      const itemTotalHeight = itemHeight + ITEM_MARGIN_BOTTOM;
+
+      const columnsHeights = columnsCumulativeTop.map((c) => c[c.length - 1]);
+      const destColumn = columnsHeights.indexOf(Math.min(...columnsHeights));
+
+      columns[destColumn].push(i);
+      columnsCumulativeTop[destColumn].push(
+        columnsHeights[destColumn] + itemTotalHeight
+      );
+    }
+
+    return [columns, columnsCumulativeTop];
+  }, [resources]);
+
+  const jsx = columns.map((column, columnIndex) => {
+    return (
+      <Column
+        key={columnIndex}
+        width={150}
+        height={
+          columnsCumulativeTop[columnIndex][
+            columnsCumulativeTop[columnIndex].length - 1
+          ]
+        }
+      >
+        {column.map((resourceIndex, i) => {
+          const resource = resources[resourceIndex];
+
+          const resized = calculateAspectRatioFit(
+            resource.width,
+            resource.height,
+            150
+          );
+
+          const top = columnsCumulativeTop[columnIndex][i];
+          const elFrom = top;
+          const elTo = top + resized.height;
+
+          const padding = 3000;
+          let isVisible =
+            elTo >= scrollInfo.from - padding &&
+            elFrom <= scrollInfo.to + padding;
+
+          const placeholderPadding = 5000;
+          let isPlaceholderVisible =
+            elTo >= scrollInfo.from - placeholderPadding &&
+            elFrom <= scrollInfo.to + placeholderPadding;
+
+          let Component = isVisible ? MediaElement : Placeholder;
```

Comment:
This is still potentially very many `divs` displayed. Especially since they are all positioned absolute?